### PR TITLE
CCDE-247: Improve reader API

### PR DESCRIPTION
**Pull request by isichei**

### CODE REVIEW

File: arrow_pd_parser/reader.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        input_file=input_file,
+        metadata=metadata,
+        ignore_columns=ignore_columns,
+        drop_columns=drop_columns,
+        pd_integer=pd_integer,
+        pd_string=pd_string,
+        pd_boolean=pd_boolean,
+        pd_date_type=pd_date_type,
+        pd_timestamp_type=pd_timestamp_type,
+        bool_map=bool_map,
```
Comment (s-ducks) [inline arrow_pd_parser/reader.py:61]: These kwargs end up in the `read_<ff>` function, of which most of them throw an unexpected kwarg error

File: githooks/pre-commit [hunk 2]
```diff
@@ -2,53 +2,6 @@
...
-while read -p "Continue with commit? (Y/n) " yn; do
-    case $yn in
-        [Yy] ) printf "Continuing with commit\n"; break;;
-        [Nn] ) printf "Cancelling commit\n"; exit 1;;
-        * ) echo "Please answer y (yes) or n (no):" && continue;
-    esac
-done
+# add .version
+poetry version | awk '{print "__version__ = \"" $2 "\""}' > arrow_pd_parser/__init__.py
+git add arrow_pd_parser/__init__.py
```
Comment (s-ducks) [inline githooks/pre-commit:7]: don't want to keep the linting and that? 🧐

**Review by s-ducks — COMMENTED** [review]
see comments

File: arrow_pd_parser/_readers.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    """
+
+    ignore_columns: List = (None,)
+    drop_columns: List = (None,)
+    pd_integer: bool = (True,)
+    pd_string: bool = (True,)
+    pd_boolean: bool = (True,)
+    pd_date_type: str = ("datetime_object",)
+    pd_timestamp_type: str = ("datetime_object",)
+    bool_map: Dict = (None,)
```
[the remark this answers is not in the document]
Comment (isichei) [reply arrow_pd_parser/_readers.py:36]: Lol I didn't add that. Thinking there might have been some python interpreter / black madness going on...

File: arrow_pd_parser/reader.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import pandas as pd
+from mojap_metadata import Metadata
+from arrow_pd_parser._readers import (
+    ArrowParquetReader,
+    PandasCsvReader,
+    PandasJsonReader,
+    get_reader_from_file_format,
+)
+from utils import infer_file_format, FileFormat
```
[the remark this answers is not in the document]
Comment (isichei) [reply arrow_pd_parser/reader.py:9]: think it should be from `arrow_pd_parser.utils` will commit

File: githooks/pre-commit (hunk 2 above)
Comment (isichei) [reply githooks/pre-commit:7]: Nope

**Pull request merged**